src/librustdoc/clean/inline.rs

```diff
@@ -427,15 +427,29 @@ fn build_module(cx: &DocContext<'_>, did: DefId, visited: &mut FxHashSet<DefId>)
         // If we're re-exporting a re-export it may actually re-export something in
         // two namespaces, so the target may be listed twice. Make sure we only
         // visit each node at most once.
-        for &item in cx.tcx.item_children(did).iter() {
-            let def_id = item.res.def_id();
-            if item.vis == ty::Visibility::Public {
-                if did == def_id || !visited.insert(def_id) {
-                    continue;
-                }
-                if let Some(i) = try_inline(cx, item.res, item.ident.name, None, visited) {
-                    items.extend(i)
-                }
+        for &item in cx.tcx.item_children(did).iter().filter(|item| item.vis == ty::Visibility::Public) {
+            let def_id = match clean::utils::res_to_def_id(cx, &item.res) {
+                Some(did) => did,
+                None => continue,
+            };
+            if did == def_id || !visited.insert(def_id) {
+                continue;
+            }
```

Comment:
As the `def_id` is only used to avoid infinite recursion, we only really care about the `def_id` of modules so using `mod_def_id()` should be all that's needed to fix the ICE:

```rust
if let Some(def_id) = item.res.mod_def_id() {
    if did == def_id || !visited.insert(def_id) {
        continue;
    }
}
```